go/Makefile

```diff
@@ -99,13 +99,13 @@ run: manifests generate fmt vet ## Run a controller from your host.
 .PHONY: test
 test:
 	CGO_ENABLED=0 go mod tidy -v
-	CGO_ENABLED=0 go test ./cli/...
+	CGO_ENABLED=0 go test -race ./cli/...
 #	CGO_ENABLED=0 go test ./controller/...
 
 .PHONY: e2e
 e2e:
-	#UPDATE_GOLDEN=true go test -skip 'TestE2E.*|TestAutogenClient' -v ./...
-	go test -skip 'TestE2E.*|TestAutogenClient' -v ./...
+	#UPDATE_GOLDEN=true go test -race -skip 'TestE2E.*|TestAutogenClient' -v ./...
+	go test -race -skip 'TestE2E.*|TestAutogenClient' -v ./...
```

Comment:
is there value for race detector in e2e? can we deploy race images to the cluster instead?

Reply:
It's a good point. I kind of see value in both:
- the race detector in e2e will ensure that if there are issues with the test code/execution, it will be flagged, right?
- the race images running in the cluster will catch runtime issues in the product

wdyt?